### PR TITLE
Create mixin to set layout widths

### PR DESCRIPTION
**Pull request by ooneill547**

### CODE REVIEW

File: src/scss/init/unity-layout/test.scss [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@import "../../init";
```
Comment (David-Corbett) [inline src/scss/init/unity-layout/test.scss]: is this file needed?

File: src/scss/_init.scss [hunk 2]
```diff
@@ -20,6 +20,7 @@
 
 @import 'init/colors';
 @import 'init/variables';
+@import 'init/mixins';
```
Comment (David-Corbett) [inline src/scss/_init.scss]: not sure you need to add this, surely it should be `unity-layout/unity-layout`

File: src/scss/2_layout/container.scss [hunk 3]
```diff
@@ -3,6 +3,8 @@
  * Defines the layouts for the container section of the page.
  */
 
+@import "../init";
```
Comment (David-Corbett) [inline src/scss/2_layout/container.scss]: don't need this @import declaration, also don't refactor the name of the file, keep it as a partial

File: src/scss/init/unity-layout/unity-layout.scss [hunk 4]
```diff
@@ -0,0 +1,69 @@
+@import "../../init";
```
Comment (David-Corbett) [inline src/scss/init/unity-layout/unity-layout.scss]: you shouldn't need to @import the init file here

File: src/scss/init/unity-layout/unity-layout.scss [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+          width: 100%;
+        }
+
+        #sidebar-second {
+          width: $value-1;
+        }
+      }
+    }
+    @else if (length($values) == 3) {
+      $value-1: nth($values, 1);
```
Comment (David-Corbett) [inline src/scss/init/unity-layout/unity-layout.scss]: need to be careful with these values as they should be set as percentage values for each column, however to get them to compile correctly we might need raw numbers, might need to do some logic around this to test what the value is, so if for example the developer forgets the % is there enough in the code to return a legitimate css rule.

**Comment by ooneill547** [comment]
I have added functions to check if the values passed into the mixin are valid, e.g. integers or rounded percentages, and that the sum of all values is less than or equal to 100.

File: src/scss/init/unity-layout/_unity-layout.scss [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+// unity-layout-values-valid($values)
+//
+// Return true if all values are valid percentages and the sum of $values is less than or equal to 100.
+//
+// Weight: -1
+//
+// Style guide: init.libraries.unity-layout.unity-layout-values-valid
+@function unity-layout-values-valid($values) {
```
Comment (David-Corbett) [inline src/scss/init/unity-layout/_unity-layout.scss]: make this a prviate scss function for this file, so declare it as _unity-layout-values-valid

File: src/scss/init/unity-layout/_unity-layout.scss [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+// is-valid($values)
+//
+// Pass each $value in $values list through convert-percentage() function. Return the new list of $values if all values
+// are valid, if not, return false.
+//
+// Weight: -3
+//
+// Style guide: init.libraries.unity-layout.is-valid
+@function is-valid($values) {
```
Comment (David-Corbett) [inline src/scss/init/unity-layout/_unity-layout.scss]: make this a private function for this scss script, also maybe give it a more semantic name to make it clear that its scope is limited to the the unity layout mixin
